File: src/versioning_manager.api/Services/OrganizationService.cs
using System.Collections.Generic;
using versioning_manager.contracts.Data;
using versioning_manager.contracts.Services;
using versioning_manager.data.Models;

namespace versioning_manager.api.Services
{
    public class OrganizationService : IOrganizationService
    {
        IOrganizationRepository _repository;
        public OrganizationService(IOrganizationRepository repository)
        {
            _repository = repository;
        }

        public IEnumerable<Organization> GetAll()
        {
            return _repository.GetAll();
        }

        public Organization Get(int id)
        {
            return _repository.Get(id);
        }

        public Organization Add(Organization org)
        {
            return _repository.Add(org);
        }
    }
}

File: src/versioning_manager.ui/src/services/OrganizationService.ts
import { Organization } from '@/models/Organization';
import Api from '@/services/Api';
import { AxiosResponse } from 'axios';

interface OrgResponse {
  id: number;
  name: string;
  description: string;
}

export default {
  async getById(id: number) {
    const response = await Api().get('/organization/' + id) as AxiosResponse<OrgResponse>;
    const organization = new Organization(response.data.id, response.data.name, response.data.description);

    return organization as Organization;
  },
  async getAll() {
    const response = await Api().get('/organization') as AxiosResponse<OrgResponse[]>;
    const orglist = response.data.map((item) => {
      return new Organization(item.id, item.name, item.description);
    });

    return orglist as Organization[];
  },
};
